### ISSUE TITLE
AIP-0123: Alternating collection identifiers and singletons

### ISSUE THREAD

**Issue by liveFreeOrCode** [googleapis/api-linter#1398]
I believe that this PR introduced a bug: https://github.com/googleapis/api-linter/pull/1151

As stated in AIP-122:

> Resource name components should usually alternate between collection identifiers (example: publishers, books, users) and resource IDs (example: 123, les-miserables, vhugo1802).

Note the word "usually"

As stated in AIP-156:

> Singleton resources **may** parent other resources.

And...

> Singleton resources **must not** have a user-provided or system-generated ID; their [resource name][aip-122] includes the name of their parent followed by one static-segment.

While, this is not explicit in the documentation, I would expect the singleton pattern, with the ability to be a parent of other resources, to take precedent and put "usually" to the test.

Curious to hear your thoughts?

**Comment by noahdietz** [comment]
Yes I think you are right - a child resource of a singleton would have sequential static segments.

What we could probably do is first lop off the last two segments e.g. `/books/{book}` and see if the remaining parent pattern exists as a resource i.e. a Singleton, and allow that case.

@liveFreeOrCode do you have any interest in working on this? 

(sorry for the late reply!)

**Comment by liveFreeOrCode** [comment]
@noahdietz yeah, I don't mind working on it, but I'm taking some PTO the next couple of weeks, so it wouldn't be until early/mid September I can look at it.
